packages/rocketchat-lib/server/methods/addUserToRoom.coffee

```diff
@@ -10,19 +10,31 @@ Meteor.methods
 			throw new Meteor.Error 'error-invalid-username', 'Invalid username', { method: 'addUserToRoom' }
 
 		room = RocketChat.models.Rooms.findOneById data.rid
+		userId = Meteor.userId()
+		user = Meteor.user()
+		userInRoom = room.usernames?.indexOf(Meteor.user().username) >= 0
```

Comment:
Should use the `user` variable here instead of making another call to `Meteor.user()`, I think..